File: 0206.reverse_linked_list/solution.java
public class ListNode {
    int val;
    ListNode next;
    ListNode(int x) { val = x; }
}

class Solution {
    public ListNode reverseList(ListNode head) {
        ListNode ans = null;
        while (head != null){
            ListNode temp = head.next;
            head.next = ans;
            ans = head;
            head = temp;
        }
        return ans;
    }
}

File: 0206.reverse_linked_list/solution.py
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None


class Solution:
    def reverse_list(self, head: ListNode) -> ListNode:
        cur, ans = head, None
        while cur:
            temp = cur.next
            cur.next = ans
            ans = cur
            cur = temp
        return ans

    def recursion_solution(self, head: ListNode, ans=None) -> ListNode:
        if not head:
            return ans
        temp = head.next
        head.next = ans
        ans = head
        head = temp
        return self.recursion_solution(head, ans)
